src/fileoverview_comment_transformer.ts

```diff
@@ -0,0 +1,80 @@
+import * as ts from 'typescript';
+import * as jsdoc from './jsdoc';
+
+const FILEOVERVIEW_COMMENTS: ReadonlySet<string> =
+    new Set(['fileoverview', 'externs', 'modName', 'mods', 'pintomodule']);
+
+function skipLeadingCommentsRecursively(n: ts.SourceFile) {
+  const start = n.getFullStart();
+  function skipLeadingComments(n: ts.Node) {
+    if (n.getFullStart() > start) return;
+    ts.setEmitFlags(n, ts.EmitFlags.NoLeadingComments);
+    ts.forEachChild(n, skipLeadingComments);
+  }
+  ts.forEachChild(n, skipLeadingComments);
+}
+
+/**
+ * A transformer that ensures the emitted JS file has an \@fileoverview comment that contains a
+ * \@suppress {checkTypes} annotation by either adding or updating an existing comment.
+ */
+export function transformFileoverviewComment(context: ts.TransformationContext) {
+  return (sf: ts.SourceFile) => {
+    let comments: ts.SynthesizedComment[] = [];
+    if (sf.statements.length) comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
```

Comment:
Why trailing comments here, rather than leading?

Reply:
It's what `transformer_util.ts` converts comments to. Added a (leading!) comment.